src/headset/oculus_mobile.c

```diff
@@ -0,0 +1,231 @@
+
+  quat_getAngleAxis(bridgeLovrMobileData.lastHeadPose.q, angle, ax, ay, az);
+}
+
+static void oculusMobileGetEyePose(HeadsetEye eye, float* x, float* y, float* z, float* angle, float* ax, float* ay, float* az) {
+  // TODO: This is very wrong!
+  oculusMobileGetPose(x, y, z, angle, ax, ay, az);
+}
+
+static void oculusMobileGetVelocity(float* x, float* y, float* z) {
+  *x = bridgeLovrMobileData.lastHeadVelocity.x;
+  *y = bridgeLovrMobileData.lastHeadVelocity.y;
+  *z = bridgeLovrMobileData.lastHeadVelocity.z;
+}
+
+static void oculusMobileGetAngularVelocity(float* x, float* y, float* z) {
+  *x = bridgeLovrMobileData.lastHeadVelocity.ax;
+  *y = bridgeLovrMobileData.lastHeadVelocity.ay;
+  *z = bridgeLovrMobileData.lastHeadVelocity.az;
+}
+
+static Controller** oculusMobileGetControllers(uint8_t* count) {
+  *count = 0; // TODO
+  return NULL;
+}
+
+static bool oculusMobileControllerIsConnected(Controller* controller) {
+  return false;
+}
+
+static ControllerHand oculusMobileControllerGetHand(Controller* controller) {
+  return HAND_UNKNOWN;
+}
+
+static void oculusMobileControllerGetPose(Controller* controller, float* x, float* y, float* z, float* angle, float* ax, float* ay, float* az) {
+}
+
+static float oculusMobileControllerGetAxis(Controller* controller, ControllerAxis axis) {
+  return 0;
+}
+
+static bool oculusMobileControllerIsDown(Controller* controller, ControllerButton button) {
+  return 0;
+}
+
+static bool oculusMobileControllerIsTouched(Controller* controller, ControllerButton button) {
+  return false;
+}
+
+static void oculusMobileControllerVibrate(Controller* controller, float duration, float power) {
+}
+
+static ModelData* oculusMobileControllerNewModelData(Controller* controller) {
+  return NULL;
+}
+
+// TODO: need to set up swap chain textures for the eyes and finish view transforms
+static void oculusMobileRenderTo(void (*callback)(void*), void* userdata) {
+  renderCallback = callback;
+  renderUserdata = userdata;
+}
+
+static void oculusMobileUpdate(float dt) {
+}
+
+HeadsetInterface lovrHeadsetOculusMobileDriver = {
+  DRIVER_OCULUS_MOBILE,
+  oculusMobileInit,
+  oculusMobileDestroy,
+  oculusMobileGetType,
+  oculusMobileGetOriginType,
+  oculusMobileIsMounted,
+  oculusMobileIsMirrored,
+  oculusMobileSetMirrored,
+  oculusMobileGetDisplayDimensions,
+  oculusMobileGetClipDistance,
+  oculusMobileSetClipDistance,
+  oculusMobileGetBoundsDimensions,
+  oculusMobileGetBoundsGeometry,
+  oculusMobileGetPose,
+  oculusMobileGetEyePose,
+  oculusMobileGetVelocity,
+  oculusMobileGetAngularVelocity,
+  oculusMobileGetControllers,
+  oculusMobileControllerIsConnected,
+  oculusMobileControllerGetHand,
+  oculusMobileControllerGetPose,
+  oculusMobileControllerGetAxis,
+  oculusMobileControllerIsDown,
+  oculusMobileControllerIsTouched,
+  oculusMobileControllerVibrate,
+  oculusMobileControllerNewModelData,
+  oculusMobileRenderTo,
+  oculusMobileUpdate
+};
+
+// Pseudo GLFW
```

Comment:
Adding note to add `platform.h` to handle platform-specific code more elegantly (not necessary for merge).

Reply:
That might also be a good time to put a more coherent abstraction over what this patch calls "NO_WINDOW".